edb/server/protocol/ai_ext.py

```diff
@@ -363,6 +378,264 @@ async def handle_request(
         return
 
 
+async def _handle_rag_request(
+    protocol: protocol.HttpProtocol,
+    request: protocol.HttpRequest,
+    response: protocol.HttpResponse,
+    db: dbview.Database,
+    tenant: srv_tenant.Tenant,
+) -> None:
+    try:
+        body = json.loads(request.body)
+        if not isinstance(body, dict):
+            raise TypeError(
+                'the body of the request must be a JSON object')
+
+        context = body.get('context')
+        if context is None:
+            raise TypeError(
+                'missing required "context" object in request')
+        if not isinstance(context, dict):
+            raise TypeError(
+                '"context" value in request is not a valid JSON object')
+
+        ctx_query = context.get("query")
+        ctx_variables = context.get("variables")
+        ctx_globals = context.get("globals")
+        ctx_max_obj_count = context.get("max_object_count")
+
+        if not ctx_query:
+            raise TypeError(
+                'missing required "query" in request "context" object')
+
+        if ctx_variables is not None and not isinstance(ctx_variables, dict):
+            raise TypeError('"variables" must be a JSON object')
+
+        if ctx_globals is not None and not isinstance(ctx_globals, dict):
+            raise TypeError('"globals" must be a JSON object')
+
+        model = body.get('model')
+        if not model:
+            raise TypeError(
+                'missing required "model" in request')
+
+        query = body.get('query')
+        if not query:
+            raise TypeError(
+                'missing required "query" in request')
+
+        stream = body.get('stream')
+        if stream is None:
+            stream = False
+        elif not isinstance(stream, bool):
+            raise TypeError('"stream" must be a boolean')
+
+        if ctx_max_obj_count is None:
+            ctx_max_obj_count = 5
+        elif not isinstance(ctx_max_obj_count, int) or ctx_max_obj_count <= 0:
+            raise TypeError(
+                '"context.max_object_count" must be an postitive integer')
```

Comment:
Not a comment for now, but I think we're going to want a lightweight HTTP API framework at some point...